internal/cli/ops_manager_clusters_indexes_create.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2020 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/convert"
+	"github.com/mongodb/mongocli/internal/description"
+	"github.com/mongodb/mongocli/internal/flags"
+	internalJson "github.com/mongodb/mongocli/internal/json"
+	"github.com/mongodb/mongocli/internal/messages"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type opsManagerClustersIndexesCreateOpts struct {
+	globalOpts
+	name            string
+	db              string
+	collection      string
+	rsName          string
+	locale          string
+	caseFirst       string
+	alternate       string
+	maxVariable     string
+	strength        int
+	caseLevel       bool
+	numericOrdering bool
+	normalization   bool
+	backwards       bool
+	unique          bool
+	sparse          bool
+	background      bool
+	keys            []string
+	options         []string
+	store           store.AutomationPatcher
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) Run() error {
+	current, err := opts.store.GetAutomationConfig(opts.ProjectID())
+	if err != nil {
+		return err
+	}
+
+	clusterConfigs := convert.FromAutomationConfig(current)
+	for _, rs := range clusterConfigs {
+		if rs.Name == opts.rsName {
+			return internalJson.PrettyPrint(rs)
+		}
+
+	}
+
+	index, err := opts.newIndexMap()
+	if err != nil {
+		return err
+	}
+
+	current.IndexConfigs = index
+
+	if err = opts.store.UpdateAutomationConfig(opts.ProjectID(), current); err != nil {
+		return err
+	}
+
+	fmt.Print(messages.DeploymentStatus(config.OpsManagerURL(), opts.ProjectID()))
+
+	return nil
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) newIndexMap() ([]*map[string]interface{}, error) {
+
+	index, err := opts.newIndex()
+
+	if err != nil {
+		return nil, err
+	}
+
+	var inInterface *map[string]interface{}
```

Comment:
I haven't found another way to get a  `[]map[string]interface{}` out of a struct.